index.js

```diff
@@ -1,10 +1,99 @@
 // import your node modules
 
 const db = require('./data/db.js');
-const server = require('express')();
+const express = require('express');
 
 // add your server code starting here
 
+const server = express();
+
+server.use(express.json());
+
+server.post('/api/posts', (req, res) => {
+
+  const post = req.body;
+
+  if (!post.title || !post.contents) {
+
+    res.status(400).json({ errorMessage: "Please provide title and contents for the post." });
+
+  }
+
+  else {
+
+    db.insert(post)
+      .then(data => res.status(201).json(data))
+      .catch(err => res.status(500).json({ error: "There was an error while saving the post to the database" }));
```

Comment:
This line looks a little bit too long; here's one option for splitting it up:
```javascript
.catch(err => res.status(500).json({
  error: "There was an error while saving the post to the database"
}));
```
or (if you're worried about clarity with all of the parens and braces):
```javascript
.catch(err => res.status(500)
  .json({
    error: "There was an error while saving the post to the database"
  })
);
```